components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/beans/DataSourceMetadata.java

```diff
@@ -16,25 +16,25 @@
 package org.wso2.carbon.datasource.core.beans;
 
 import org.wso2.carbon.datasource.utils.DataSourceUtils;
-
-import javax.xml.bind.annotation.XmlElement;
-import javax.xml.bind.annotation.XmlRootElement;
-import javax.xml.bind.annotation.XmlType;
+import org.wso2.carbon.kernel.annotations.Configuration;
+import org.wso2.carbon.kernel.annotations.Element;
 
 /**
  * This class represents data source meta information.
  */
-@XmlRootElement(name = "datasource")
-@XmlType(propOrder = {"name", "description", "jndiConfig", "definition"})
+@Configuration(description = "Data Source metadata configuration")
```

Comment:
The hashCode method in this class simply returns -1. IMO this method should return an integer-specific to the object instance. (http://stackoverflow.com/questions/3563847/what-is-the-use-of-hashcode-in-java) Is there a specific reason for returning -1 in this method